Revise English

tajmone opened the pull request:
Polish the English text in the various markdown
documents, fixing typos and markdown formatting
issues, and render the text more fluid to read.

SicroAtGit reviewed (changes requested):
Thanks a lot for the PR!

There are a few things that should be adjusted.

Note that changes in command descriptions in `README.md` should also be adjusted in code comments in `RegExEngine.pbi` and `DfaMatcher.pbi`. The code comments should preferably not exceed the line column count of 80. Sorry for the double work. A dev tool that synchronizes or extracts the command descriptions I already have on my to-do list.

tajmone commented on README.md:
```diff
@@ -55,34 +55,34 @@ Else
-| `\*`    | Escapes the metacharacter `*` to use it as a normal character.<br>Works also with the other metacharacters: `\|` `+` `?` `(` `)` `\` |
+| `x*`    | Zero or more consecutive `x` |
+| `x+`    | One or more consecutive `x` |
+| `x?`    | Zero or one consecutive `x` |
```
What do you mean "makes no sense"? The 'x's have to be consecutive. I've added it to clarify that it doesn't mean zero or more Xs within the string.

tajmone replied:
Sorry, I now see what you meant! It can only be one at most.

I really missed that.

tajmone commented on README.md:
```diff
@@ -55,34 +55,34 @@ Else
 | `\t`    | Matches the horizontal tab character (`\x09`) |
 | `\f`    | Matches the form feed character (`\x0C`) |
-| `[x]`   | `x` can be several mixes of: single character, escape sequence or range (`a-c`) |
+| `[x]`   | Where `x` can be a combination of: single literal character, escape sequence, or literals range (`a-c`) |
```
I thought that ranges applies only to literal characters, not escape sequences, but if the latter is also the case then it definitely needs adjusting. The problem is that there isn't much space really in markdown table cells.

tajmone replied:
I've changed "or literals range" to simply "or range" — after all, this table is mostly for guideline, it doesn't need to be too specific, but at least it shouldn't be ambiguous or imprecise.

tajmone commented:
> Note that changes in command descriptions in README.md should also be adjusted in code comments in RegExEngine.pbi and DfaMatcher.pbi. The code comments should preferably not exceed the line column count of 80. 

Done! Should be all OK. I've also polished a couple of extra comments that I remembered needed some tweaking.

> A dev tool that synchronizes or extracts the command descriptions I already have on my to-do list.

If you use AsciiDoc instead of markdown there's nothing extra that you need to do, just [mark the text-regions via tag-comments], and Asciidoctor will extract them for you at conversion time. Removing the comment delimiters then is fairly simple using Ruby, e.g. directly within the project Rakefile.

[mark the text-regions via tag-comments]: https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/ "AsciiDoc Manual » Include Content by Tagged Regions"

tajmone commented on README.md:
```diff
@@ -248,8 +248,20 @@ Runs the DFA against the string. The function requires the pointer to the string
+[Alphabetic, M, Nd, Pc and Join_Control]: https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3AAlphabetic%3A%5D%5B%3AM%3A%5D%5B%3ANd%3A%5D%5B%3APc%3A%5D%5B%3AJoin_Control%3A%5D&abb=on&esc=on&g=&i= "UnicodeSet Utility: characters classes 'Alphabetic', 'M', 'Nd', 'Pc' and 'Join_Control'"
+
+[MIT license]: ./LICENSE
+[CONRTIBUTING]: ./CONTRIBUTING.md
```
Done!

SicroAtGit commented on README.md:
```diff
@@ -248,8 +248,20 @@ Runs the DFA against the string. The function requires the pointer to the string
 
```
Please add the same changes from the main module here as well:

> Runs the DFA against the target string, passed via a pointer. The match search will start from the beginning of the string. If a match is found, the byte length of the match is returned, otherwise null. If the address of an integer variable was passed as the optional `*regExId` parameter, the RegEx ID number of the matching RegEx is written into it. If multiple RegExes match the same string, each having been assigned a different RegEx ID number, the RegEx ID number of the last matching RegEx will be picked, i.e. the matching RegEx that was last added with the `AddNfa()` function.

(I have adjusted the quoted text, by replacing "RegEx engine" with "DFA".)

The text should also be adjusted in `DfaMatcher.pbi`.

SicroAtGit replied:
Sorry, if it was misleading, but I meant it that way:

The description of the `Match()` of the `DfaMatcher` _only_ in line 247 should be changed from

> Runs the DFA against the string. The function requires the pointer to the string. The match search will start from the beginning of the string. If a match is found, the byte length of the match is returned, otherwise null. If an address to an integer variable was passed in the optional `*regExId` parameter, the RegEx ID number of the matched RegEx is written into it. If there are multiple RegExes that match the same string and have been assigned different RegEx ID numbers, the RegEx ID number of the last matched RegEx is taken, i.e. the last matched RegEx added with the `AddNfa()` function from the main module.

to

> Runs the DFA against the target string, passed via a pointer. The match search will start from the beginning of the string. If a match is found, the byte length of the match is returned, otherwise null. If the address of an integer variable was passed as the optional `*regExId` parameter, the RegEx ID number of the matching RegEx is written into it. If multiple RegExes match the same string, each having been assigned a different RegEx ID number, the RegEx ID number of the last matching RegEx will be picked, i.e. the matching RegEx that was last added with the `AddNfa()` function.

SicroAtGit commented on README.md:
```diff
@@ -239,7 +239,7 @@ EndStructure
 - **`GetString(_memoryAddress_, _lengthInBytes_)`**<br><br>
-Simplifies the return of the match as a string.
+Simplifies extracting the matched string via its memory address and length info obtained from a `Match()` call.
 
```
The text should also be adjusted in `DfaMatcher.pbi`.

SicroAtGit commented on README.md:
```diff
@@ -184,27 +184,27 @@ Simplifies the return of the match as a string.
 
 - **`Match(*regExEngine.RegExEngineStruc, *string.Unicode, *regExId.Integer = 0)`**<br><br>
-Runs the RegEx engine against the string. The function requires the pointer to the string. The match search will start from the beginning of the string. If a match is found, the byte length of the match is returned, otherwise null. If an address to an integer variable was passed in the optional `*regExId` parameter, the RegEx ID number of the matched RegEx is written into it. If there are multipl ...
+Runs the DFA against the target string, passed via a pointer. The match search will start from the beginning of the string. If a match is found, the byte length of the match is returned, otherwise null. If the address of an integer variable was passed as the optional `*regExId` parameter, the RegEx ID number of the matching RegEx is written into it. If multiple RegExes match the same string, each ...
```
```suggestion
Runs the RegEx engine against the target string, passed via a pointer. The match search will start from the beginning of the string. If a match is found, the byte length of the match is returned, otherwise null. If the address of an integer variable was passed as the optional `*regExId` parameter, the RegEx ID number of the matching RegEx is written into it. If multiple RegExes match the same string, each having been assigned a different RegEx ID number, the RegEx ID number of the last matching RegEx will be picked, i.e. the matching RegEx that was last added with the `AddNfa()` function.
```

tajmone replied:
Done, I then squashed your proposed commit, so we keep the PR as a single commit.

tajmone committed:
Revise English

Polish the English text in the various markdown
documents, fixing typos and markdown formatting
issues, and render the text more fluid to read.

Amend source code comments accordingly, so that
descriptions of procedures in the library module
matches the changes in the README document.

SicroAtGit commented:
A few things got mixed up, which I have now fixed.

SicroAtGit commented:
@tajmone

Thanks again for the PR.

> If you use AsciiDoc instead of markdown there's nothing extra that you need to do, just [mark the text-regions via tag-comments](https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/), and Asciidoctor will extract them for you at conversion time. Removing the comment delimiters then is fairly simple using Ruby, e.g. directly within the project Rakefile.

Thanks, I will have a look. Editing the documentation really needs to become easier without having to edit the same texts in multiple places. I will think about it and then open a discussion.
